opm/simulators/flow/Main.hpp

```diff
@@ -433,18 +433,10 @@ namespace Opm
 
                 Opm::FlowMainEbos<PreTypeTag>::printPRTHeader(outputCout_);
 
-                if (outputCout_) {
-                    OpmLog::info("Reading deck file '" + deckFilename + "'");
-                }
-
                 readDeck(mpiRank, deckFilename, deck_, eclipseState_, schedule_,
                          summaryConfig_, nullptr, python, std::move(parseContext),
                          init_from_restart_file, outputCout_);
 
-                if (outputCout_) {
-                    OpmLog::info("Done reading deck file.");
-                }
```

Comment:
Actually, this message is really useful.  It was added in #2760 with express intention that users get some idea of what part of the setup process is going on.

Reply:
OK; we are in the process of adding more informative messages which will give essentially the same information. Can of course let this message remain in the interim.

Reply:
> we are in the process of adding more informative messages which will give essentially the same information

Right, I got that impression from other work being submitted at the moment.

> Can of course let this message remain in the interim.

I guess that depends on the timing.  If the new system is expected to be in place in a few days then there's no need.  Otherwise I'd prefer to keep this message.

Reply:
I have resurrected the message

Reply:
Actually, I beg to differ. The message is wrong. The end of parsing is already reported where it actually happens. Keeping this message has as such a negative value. I totally agree that the user needs a fine grained control of what happens after parsing, but this message is not helping that.

Reply:
> The end of parsing is already reported where it actually happens

I did not think so. Where?

Reply:
All lines parsed are reported, including the last. For Norne it looks like this:
```
 1380 Reading WCONINJE in /home/rustad/Dev/Data/opm-data/norne/INCLUDE/BC0407_HIST01122006.SCH line 9554
 1381 Reading DATES    in /home/rustad/Dev/Data/opm-data/norne/INCLUDE/BC0407_HIST01122006.SCH line 9563

Warning: Unsupported value for ENDSCALE
```

Reply:
If you know that is the last line, then you are good. But most users do not know that, and if something goes wrong after this point (but before timesteps start) they could blame the deck reader/parser wrongly. That actually happened, which is why we added the extra (user-focused) information that we are done with it now.

Reply:
Seriously, it is always the last line, by design. Just run Flow on some cases and you immediately see that. If something goes wrong after this point we need as I already stated (and which has been discussed and agreed by both @joakim-hove and @bska ) fine grained control on what happens. Basically we need logging of where between the parsing and time step start we are. I repeat, the message in question is not helping that, it is simply wrong.

Reply:
> Basically we need logging of where between the parsing and time step start we are.

I do not disagree.

> I repeat, the message in question is not helping that, it is simply wrong.

This is the code snippet:
```
                readDeck(mpiRank, deckFilename, deck_, eclipseState_, schedule_,
                         summaryConfig_, nullptr, python, std::move(parseContext),
                         init_from_restart_file, outputCout_);

                if (outputCout_) {
                    OpmLog::info("Done reading deck file.");
                }
```
If this is wrong, then what is perhaps wrong is that readDeck() does more than just reading the deck, it also creates the EclipseState and calls some checks. Is that what you mean? If so, we agree!  I now believe the printout should be moved into readDeck(), after the deck is constructed but before the checks are called (line 192 of readDeck.cpp).

In general, I find that a little bit of information to the user about what phase in the process we have reached is good. Do we agree on that?

Reply:
> Seriously, it is always the last line, by design. Just run Flow on some cases and you immediately see that

I do not understand what you mean by this?

Reply:
> Just run Flow on some cases and you immediately see that. If something goes wrong after this point we need

The one thing we **don't** detect if we remove this message is if something goes wrong **before** this point&mdash;e.g., we're reading from a network disk and the controller died meaning the process then just hangs with no additional clue to the problem.  It **could be** a problem constructing the parser output object (`EclipseState`, `EclipseGrid`, `Schedule`, or `SummaryConfig`) or just read failure.

I suppose we could output one final record of the form
```
1234 END (Implied)
```
when we actually get to the end of the input file.  If the input happens to contain an `END` keyword then we don't output that and don't output the "implied" end record.

Reply:
> If this is wrong, then what is perhaps wrong is that readDeck() does more than just reading the deck, it also creates the EclipseState and calls some checks. Is that what you mean? If so, we agree!

Yes, this is exactly what I mean. We need the EclipseState to report what it does when it does it, so we can have a clue as to what goes wrong when it crashes.

> 1234 END (Implied)

That is a possibility, but I am afraid we are solving a problem that does not exist. The user always has the deck available, and will check the deck if Flow crashes during parsing. That was exactly why we needed that keyword-by-keyword reporting, and afaik it does the job.